### ISSUE TITLE
Shurikenpi.io is not running / not installing with sudo

### ISSUE THREAD

**Issue by checkuon** [Urban-Hacker/shurikenpi.io#3]
I followed the instructions detailed here on Github and on the website to make the script executable and run it. On the screenshot below you can see the first part of the output:
![Shurikenpi install](https://github.com/Urban-Hacker/shurikenpi.io/assets/94130132/01764356-1541-43b0-b88f-76f80465b112)
As you can see it first warns that I'm not root, but then it follows up by saying 'We are root :)'

If I go into ~/shurikenpi/shurikenpi.io directory and run shuriken.sh I get the following output: 
![Running shurikenpi](https://github.com/Urban-Hacker/shurikenpi.io/assets/94130132/4df3220a-558a-473a-a1bc-8648200e7bd7)
Running the shuriken.sh script with sudo gives the same result.

I then deleted everything and ran the install script with sudo, here is the ouput:
![Shuriken sudo install](https://github.com/Urban-Hacker/shurikenpi.io/assets/94130132/c57603b4-b68f-470c-a749-88ba318c38e5)

The installation does seem to complete but if I then run 'ls -la' my home directory it doesn't show the shurikenpi directory:
![List home directory](https://github.com/Urban-Hacker/shurikenpi.io/assets/94130132/ef0bf41c-56bb-42a8-a4d1-a4cc7a42da46)

**Comment by Urban-Hacker** [comment]
Hi, it seems the installation is okay but it fails when trying to start it. Can you tell me what is the content of: 

shurikenpi/shurikenpi.io/Worker/worker.sh

Be aware, this might contain your btc address, so please redact it. You can also start the worker manually with ./worker.sh. Let me know what it says.

In the meantime I will add an option to start it from CMD without doing + better debug. I’ll do another version in the coming days.

**Comment by checkuon** [comment]
../Bin/arm64_cpuminer -a sha256d -o stratum+tcp://solo.ckpool.org:3333 -u bc1qs9suund5dx4l3ng363fx5f3ef7ulygaw93epsk

I put in a dummy address. I had to install without sudo obviously or there was no ~/shurikenpi/shurikenpi.io directory at all.

**Comment by Urban-Hacker** [comment]
Strange, it seems the worker is properly created. If you run the worker? Does it starts at all?

**Comment by checkuon** [comment]
file:///home/vincent/Pictures/Screenshots/Screenshot%20from%202024-01-26%2021-45-13.png

**Comment by checkuon** [comment]
![Screenshot from 2024-01-26 21-45-13](https://github.com/Urban-Hacker/shurikenpi.io/assets/94130132/54a525d5-8bba-4f51-a23e-387a1772f1e5)

going into /Bin it does appear to be there:
![Screenshot from 2024-01-26 21-47-23](https://github.com/Urban-Hacker/shurikenpi.io/assets/94130132/fb618c6f-6b93-4aa4-b3b7-cccce413bf7a)

tried running it 
![Screenshot from 2024-01-26 21-49-38](https://github.com/Urban-Hacker/shurikenpi.io/assets/94130132/384fcf70-83af-4926-8954-de6c07492926)

**Comment by cast4way** [comment]
I can confirm, having the exact same issue as checkuon.

**Comment by Urban-Hacker** [comment]
Found the problem. A fix will come soon. Thanks to @cast4way to help me debug this problem.

**Comment by Urban-Hacker** [comment]
Can you both try again with the newest install script? Just wget the new script and let me know if it works. I think I added all the required dependencies + fixed some bugs in the installer.

@checkuon & @cast4way

**Comment by cast4way** [comment]
![Screenshot from 2024-02-18](https://github.com/Urban-Hacker/shurikenpi.io/assets/83720713/967bf1eb-f653-44c2-918a-e309f28469eb)
For me it fails to install the lib dependencies and shuriken folder is not created.

**Comment by Urban-Hacker** [comment]
Which raspberry pi version are you using and which OS are you using, with which architecture? 64 bits vs 32 bits etc.

**Comment by cast4way** [comment]
64 bit ubuntu server 23.10 for pi. On a pi 4 B.

**Comment by cast4way** [comment]
Pi 4B

**Comment by Urban-Hacker** [comment]
Could you try on raspbian?

**Comment by cast4way** [comment]
Sure, will let you know later today.

**Comment by Urban-Hacker** [comment]
Edit: Installer is updated. If you re-download the latest installer it should work 

Never mind do not try on raspbian, I forgot to add something.  Can you run

sudo dpkg --add-architecture armhf

Then do it again? If it works I will update the installer then

**Comment by cast4way** [comment]
Getting closer, I did the following then got an install error for **libssl1.1:armhf** 

`sudo dpkg --add-architecture armhf`
`rm -rf install.sh`
`wget https://raw.githubusercontent.com/Urban-Hacker/shurikenpi.io/main/install.sh`
`chmod +x ./install.sh`
`./install.sh`

> shuri@ken:~$ ./install.sh
> 
>   » Root user check...
> 
>     ShurikenPi called with non-root priviledeges :(
>     Elevated priviledeges are required to install and run ShurikenPi
>     Please check the installer for any concerns about this requirement
>     Make sure you downloaded this script from a trusted source
> 
>   » Correct password.
>   » We are root :)
> 
>   » The next steps might take some time, be patient
>   » Installing pre-requisites...
>   » Update cache...
>   » apt install gum                                                           ✓
>   » apt install git                                                           ✓
>   » apt install tor                                                           ✓
>   » apt install curl                                                          ✓
>   » apt install screen                                                        ✓
>   » dpkg --add-architecture armhf                                             ✓
>   » apt install libc6:armhf                                                   ✓
>   » apt install libstdc++6:armhf                                              ✓
>   » apt install libcurl4:armhf                                                ✓
>   » apt install libssl1.1:armhf                                               ✓ [FATAL ERROR]
>   » --- BEGIN ERROR DUMP ---
>     Reading package lists...
>     Building dependency tree...
>     Reading state information...
>     E: Unable to locate package libssl1.1:armhf
>     E: Couldn't find any package by glob 'libssl1.1'
>     E: Couldn't find any package by regex 'libssl1.1'
>   » --- END ERROR DUMP ---

**Comment by Urban-Hacker** [comment]
If you do sudo apt-get install libssl3:armhf ?

But I think it might be a smart move to try on raspbian. If that works on raspbian I’m going to close the ticket, add a check to make sure we run on raspbian and call it a day.

* So can you try first (on your ubuntu server): sudo apt-get install libssl3:armhf 
* If that does not work just install raspberry pi OS, run the script and let me know.

Sadly I’m traveling so I do not have myself access to a r-pi to do all those tests myself :s

**Comment by cast4way** [comment]
I installed it on rasbpi OS, installation worked, but two issues remain:
1. `sudo apt-get install libjansson4:armhf` is missing in installation script
2. shurikan start cmd from home does not work

```
ken@barbie:~ $ shuriken help
-bash: shuriken: command not found
ken@barbie:~ $ cd shurikenpi/
ken@barbie:~/shurikenpi $ ls
shurikenpi.io
ken@barbie:~/shurikenpi $ cd shurikenpi.io/
ken@barbie:~/shurikenpi/shurikenpi.io $ ./shuriken.sh start
  » Starting up shurikenpi.io...
  » shurikenpi.io is not running
ken@barbie:~/shurikenpi/shurikenpi.io $ ./shuriken.sh test
  » Starting interal self-test
  » Internet is accessible.
  » Running the mining process for 1 min
  ! Please do not force stop the process during the test
../Bin/arm64_cpuminer: error while loading shared libraries: libjansson.so.4: cannot open shared object file: No such file or directory
  » Self Test Finished!
ken@barbie:~/shurikenpi/shurikenpi.io $ cd
ken@barbie:~ $ sudo apt-get install libjansson4:armhf
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
The following NEW packages will be installed:
  libjansson4:armhf
0 upgraded, 1 newly installed, 0 to remove and 0 not upgraded.
Need to get 36.7 kB of archives.
After this operation, 81.9 kB of additional disk space will be used.
Get:1 http://deb.debian.org/debian bookworm/main armhf libjansson4 armhf 2.14-2 [36.7 kB]
Fetched 36.7 kB in 0s (159 kB/s)         
Selecting previously unselected package libjansson4:armhf.
(Reading database ... 145255 files and directories currently installed.)
Preparing to unpack .../libjansson4_2.14-2_armhf.deb ...
Unpacking libjansson4:armhf (2.14-2) ...
Setting up libjansson4:armhf (2.14-2) ...
Processing triggers for libc-bin (2.36-9+rpt2+deb12u4) ...
ken@barbie:~ $ shuriken help
-bash: shuriken: command not found
ken@barbie:~ $ cd ~/shurikenpi/shurikenpi.io
ken@barbie:~/shurikenpi/shurikenpi.io $ ./shuriken.sh start
  » Starting up shurikenpi.io...
  » shurikenpi.io is running
```

**Comment by checkuon** [comment]
I actually can't remember if I'm on Raspberry Pi OS or Ubuntu but it looks to be Ubuntu from this output. I'll try with Raspberry Pi OS.

![Screenshot from 2024-02-20 22-57-11](https://github.com/Urban-Hacker/shurikenpi.io/assets/94130132/3125dfc8-baf7-4a40-b759-f69a3acbce4a)

**Comment by ghost** [comment]
<img width="567" alt="Screenshot 2024-03-14 at 16 27 22" src="https://github.com/Urban-Hacker/shurikenpi.io/assets/107030871/c0e4db55-291b-4f2c-a865-2b56fd6d0a58">

I am having the same problem with the miner not running. On a rasp4 with raspiOS 64 bit. Ran a self test ans see this.

**Comment by ghost** [comment]
RAN sudo apt-get install libjansson4:armhf 

Updated and upgraded and miner ran!

**Comment by checkuon** [comment]
@chef4brains I really want to get this working. I will reflash my sd card and start over. Did you run the install.sh script with sudo?

**Comment by cast4way** [comment]
@checkuon start over on raspi OS, I do not think you need to run the install.sh with sudo, but it makes sense to do so.
Then have a look what I posted on feb 18. 
1.
`sudo apt-get install libjansson4:armhf`
2.
`cd shurikenpi.io/`
3.
`./shuriken.sh start`

**Comment by ghost** [comment]
> @chef4brains I really want to get this working. I will reflash my sd card and start over. Did you run the install.sh script with sudo? 

I didn't need to use sudo. I was logged in as root from the beginning of the install.

**Comment by Urban-Hacker** [comment]
Okay I found out what is some of the issues (and thanks for your support / help) proving logs etc to help me understand:
* I forgot to refresh the apt cache after enabling special libraries required. I mostly test it on a virtual machine on my computer so of course this was never triggered.
* And when you run it from sudo (sudo ./install.sh) this will install it in the root directory, not the user directory. Of course when you attempt running it afterwards it fails.

I already fixed the refresh and I’ll think how to fix the sudo issue
